onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -173,12 +230,13 @@ Status Conv<T>::ComputeInternal(OpKernelContext* context) const {
     }
 
     if (!y_data) {
-      Tensor* Y = context->Output(0, TensorShape(s_.y_dims));
+      Y = context->Output(0, TensorShape(s_.y_dims));
       // special case when there is a dim value of 0 in the shape.
       if (Y->Shape().Size() == 0)
         return Status::OK();
 
-      y_data = reinterpret_cast<CudaT*>(Y->template MutableData<T>());
+      y_data = !s_.post_slicing_required ? reinterpret_cast<CudaT*>(Y->template MutableData<T>())
+                                         : reinterpret_cast<CudaT*>(s_.cached_memory_for_cudnn_results);
     }
```

Comment:
If y_data is nullptr is s_.cached_memory_for_cudnn_results guaranteed to have been allocated and of the correct size? 

It looked like y_data would be set by AllocMemoryForCuDnnConvResults if an allocation happened, and nullptr if it had not.

Reply:
yes s_.cached_memory_for_cudnn_results is guaranteed to have been allocated for now AFAIK. 

We enter this `if` for a "subsequent" run awhere the input shape and the filter shape was the same as the previous run, (i.e.) output shape is also going to be the same. Since, we enter this `if`, we have seen another Run() and we would have allocated memory then.

If this was the first Run(), we would have allocated (and cached for subsequent runs) `y_data` ahead (and hence we wouldn't be entering this `if`).

But this entire debate is moot, if we have to ditch the "allocate and cache" memory approach and allocate memory for every Run() which makes it more simpler actually.